components/gfx/display_list/mod.rs

```diff
@@ -148,6 +148,58 @@ impl DisplayList {
         }
         result
     }
+
+    // Print the display list. Only makes sense to call it after performing reflow.
+    pub fn print_items(&self, mut indentation: String) {
+        let min_length = 4;
+        // We cover the case of an empty string.
+        if indentation.len() == 0 {
+            indentation = String::from_str("####");
+        }
+
+        // We grow the indentation by 4 characters if needed.
+        if indentation.len() < min_length {
```

Comment:
I think this can just be:

    while indentation.len() < min_length {
        let c = indentation.char_at(0);
        indentation.push(c);
    }